pkg/pool-manager/virtualmachine_pool.go

```diff
@@ -458,7 +455,10 @@ func (p *PoolManager) MarkVMAsReady(vm *kubevirt.VirtualMachine, parentLogger lo
 	p.poolMutex.Lock()
 	defer p.poolMutex.Unlock()
 
+	logger.V(1).Info("updating macpoolMap according to vmToMacPoolMap", "vm's vmToMacPoolMap", p.vmToMacPoolMap[vmNamespaced(vm)])
+	var releases map[string]string
 	err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		releases = map[string]string{}
```

Comment:
if I create `releases` in the  `RetryOnConflict`context, then it will be lost outside of that scope.

Because the configmap is scanned according to the `VmToMacPoolMap` state, if I update  `VmToMacPoolMap` in the  `RetryOnConflict` context and the updates fails on conflict, then the second try will not be consistent..

For that reason, I want the `VmToMacPoolMap` changes to persist only if the configMap is successfully updated, and that releases will be cleaned on each retry...

Reply:
Oh of course! Missed it